app/src/main/java/me/timpushkin/sgbmandroidapp/utils/StorageUtils.kt

```diff
@@ -0,0 +1,48 @@
+package me.timpushkin.sgbmandroidapp.utils
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.net.Uri
+import android.util.Log
+import com.google.modernstorage.storage.AndroidFileSystem
+import com.google.modernstorage.storage.toOkioPath
+import java.io.File
+import java.io.FileNotFoundException
+import okio.IOException
+import okio.Path.Companion.toOkioPath
+
+private const val TAG = "StorageUtils"
+
+class StorageUtils(private val context: Context) {
+    private val mFileSystem = AndroidFileSystem(context)
+
+    fun Uri.copyToCache(filename: String): File? =
+        try {
+            File(context.cacheDir, filename).also { copy ->
+                mFileSystem.copy(toOkioPath(), copy.toOkioPath())
+            }
+        } catch (e: IOException) {
+            Log.e(TAG, "Failed to copy $this to $filename", e)
+            null
+        }
+
+    fun ByteArray.writeToCache(filename: String): File = File(context.cacheDir, filename).apply {
+        writeBytes(this@writeToCache)
+    }
+
+    fun Bitmap.writeToCache(filename: String): File? {
+        val file = File(context.cacheDir, filename)
+
+        val success = try {
+            file.outputStream().use { out -> compress(Bitmap.CompressFormat.PNG, 100, out) }
+        } catch (e: FileNotFoundException) {
+            Log.e(TAG, "Failed to write a bitmap to $filename", e)
+            false
+        }
+
+        return if (success) file else null.also { file.delete() }
```

Comment:
why so long, why not to return in try and delete in catch, null return after try catch

Reply:
Images caching is gone, so is this piece of code